src/page/class.rs

```diff
@@ -1,16 +1,27 @@
 // SPDX-License-Identifier: Apache-2.0
 
-/// The class of a page
-///
-/// This enumeration identifies the kind of one or more pages. This type is
-/// named `PAGE_TYPE` in the Intel documentation.
+/// The type of an enclave page (see Intel SDM Volume 3D section 34.12.2).
+/// Enclave Page Cache Map (EPCM) hols this information for each valid enclave
+/// page.
 #[repr(u8)]
 #[non_exhaustive]
 #[derive(Copy, Clone, Debug, PartialEq)]
 pub enum Class {
+    /// SGX Enclave Control Structure (SECS)
     Secs = 0,
+    /// Thread Control Structure (TCS)
     Tcs = 1,
-    Reg = 2,
-    Va = 3,
-    Trim = 4,
+    /// Regular page
+    /// FIXME: Reg -> Regular
+    Reg = 2, // deprecated in 0.4
+    /// Version Array (VA) page
+    /// FIXME: Va -> VersionArray
+    Va = 3, // deprecated in 0.4
+    /// Removable from a running enclave
+    /// FIXME: Trim -> Trimmed
+    Trim = 4, // deprecated in 0.4
+    /// The first page of a shadow stack
+    ShadowStackFirst = 5,
+    /// A shadow stack page
+    ShadowStackRest = 6,
 }
```

Comment:
We aren't super concerned about API stability in this crate yet. So instead of the `FIXME` for renames, just do the rename.

Reply:
OK, right, I can then include the corresponding change to my PR in enarx that does SGX2 things. Just did not want to break things.

Reply:
Don't worry about breaking things (yet).